### PR TITLE
Update matrix-sdk to implement devices stream

### PR DESCRIPTION
**Pull request by bitfriend**
Will upgrade `matrix-sdk` to the latest revision (2023-09-28).

### CODE REVIEW

**Comment by bitfriend** [comment]
Timeline introduced new event `m.poll.start`.
So updated both rust side and flutter side for it.

File: native/acter/src/api/message.rs [hunk 1]
```diff
@@ -1252,10 +1246,7 @@ impl RoomMessage {
             event.origin_server_ts.get().into(),
             "m.room.name".to_string(),
         );
-        let body = match event.content.name {
-            Some(name) => format!("changed name to {name}"),
-            None => "changed name".to_string(),
-        };
+        let body = format!("changed name to {}", event.content.name);
```
Comment (gnunicorn) [inline native/acter/src/api/message.rs:1249]: is `event.content.name` not an `Option<String>` anymore?

We changed this so that the format in the timeline isn't those awkward `changed name to Some('new name')` anymore ... If this is still an option please revert this change.

File: native/acter/src/api/message.rs [hunk 2]
```diff
@@ -1270,10 +1261,7 @@ impl RoomMessage {
...
```
Comment (gnunicorn) [inline native/acter/src/api/message.rs:1264]: same as above.

File: native/acter/src/api/message.rs [hunk 3]
```diff
@@ -1583,16 +1571,16 @@ impl RoomMessage {
...
-            Some(via) => format!(
-                "changed parent to {}",
-                via.iter()
-                    .map(|x| x.to_string())
-                    .collect::<Vec<String>>()
-                    .join(", "),
-            ),
-            None => "".to_string(),
-        };
+        let body = format!(
```
Comment (gnunicorn) [inline native/acter/src/api/message.rs:1574]: same as above

File: native/acter/src/api/message.rs [hunk 4]
```diff
@@ -1608,16 +1596,16 @@ impl RoomMessage {
...
```
Comment (gnunicorn) [inline native/acter/src/api/message.rs:1599]: and as above.

File: native/acter/src/api/room.rs [hunk 5]
```diff
@@ -645,7 +640,7 @@ impl Room {
             .await?
     }
 
-    pub async fn set_name(&self, name: Option<String>) -> Result<OwnedEventId> {
+    pub async fn set_name(&self, name: String) -> Result<OwnedEventId> {
```
Comment (gnunicorn) [inline native/acter/src/api/room.rs:643]: we can't unset the name anymore? That sounds like the API should still allow for it ...

File: native/acter/src/api/stream.rs [hunk 6]
```diff
@@ -112,9 +109,7 @@ impl TimelineStream {
                 let mut edited_content = RoomMessageEventContent::text_markdown(new_msg);
                 edited_content.relates_to = Some(Relation::Replacement(replacement));
 
-                timeline
-                    .send(edited_content.into(), txn_id.as_deref().map(Into::into))
-                    .await;
+                timeline.send(edited_content.into()).await;
```
Comment (gnunicorn) [inline native/acter/src/api/stream.rs:112]: didn't we say we want to replace the timeline.send for edit events with the appropriate [`timeline.edit`](https://matrix-org.github.io/matrix-rust-sdk/matrix_sdk_ui/timeline/struct.Timeline.html#method.edit) ?

**Review by gnunicorn — APPROVED** [review]
Good job. Just a few minor remarks.

**Comment by gnunicorn** [comment]
@bitfriend what's still missing to merge this?

File: native/acter/src/api/message.rs (hunk 1 above)
Comment (bitfriend) [reply native/acter/src/api/message.rs:1275]: `event.content.name` is not an `Option<String>` anymore.

File: native/acter/src/api/message.rs (hunk 2 above)
Comment (bitfriend) [reply native/acter/src/api/message.rs:1290]: `event.content.name` is not an `Option<String>` anymore.

File: native/acter/src/api/room.rs (hunk 5 above)
Comment (bitfriend) [reply native/acter/src/api/room.rs:643]: `matrix_sdk::room::Room::set_name()` doesn't accept `Option<String>`.
